ShaderTool/Command/Map.cs

```diff
@@ -187,6 +187,20 @@ class Map {
             resourceStream.Write(BitConverter.GetBytes(TGR_VERSION));
 
             // TODO
+            MapData mapData = Load(mapName);
+            string[] mapTextureName = mapData.textureNames;
+            for (int i = 0; mapTextureName.Length < 0; i--)
```

Comment:
```suggestion
            for (int i = 0; mapTextureName.Length < 0; i++)
```
This loop begins from 0 and goes to `mapTextureName.Length` so you need to increment the counter not decrement it

Reply:
Further more the condition should be the other way around you need to test if i is smaler then the length